src/core/lib/iomgr/resource_quota.cc

```diff
@@ -430,39 +451,53 @@ static bool rq_reclaim(grpc_resource_quota* resource_quota, bool destructive) {
  * ru_slice: a slice implementation that is backed by a grpc_resource_user
  */
 
-typedef struct {
+static void ru_slice_ref(void* p);
+static void ru_slice_unref(void* p);
+static const grpc_slice_refcount_vtable ru_slice_vtable = {
+    ru_slice_ref, ru_slice_unref, grpc_slice_default_eq_impl,
+    grpc_slice_default_hash_impl};
+
+namespace {
+struct ru_slice_refcount {
+  ru_slice_refcount(grpc_resource_user* resource_user, size_t size)
+      : resource_user(resource_user), size(size) {
+    base.vtable = &ru_slice_vtable;
+    base.sub_refcount = &base;
+  }
+  ~ru_slice_refcount() {
+    grpc_resource_user_free(resource_user, size);
+  }
+
+  void Ref() { refs.Ref(); }
+  void Unref() {
+    if (!refs.Unref()) {
+      return;
+    }
+    this->~ru_slice_refcount();
+    gpr_free(this);
```

Comment:
Why not use `grpc_core::Delete<>` here?

Reply:
Please see my comment below. This cannot created with `grpc_core::New()`.